specs/span-batches.md

```diff
@@ -149,11 +150,16 @@ Span-batch rules, in validation order:
   - Rules:
     - `start_epoch_num + sequence_window_size < inclusion_block_number` -> `drop`:
       i.e. the batch must be included timely.
-    - `end_epoch_num < epoch.number` -> `future`: i.e. all referenced L1 epochs must be there.
-    - `end_epoch_num == epoch.number`:
-      - If `batch.l1_origin_check != epoch.hash[:20]` -> `drop`: verify the batch is intended for this L1 chain.
-    - `end_epoch_num > epoch.number` -> `drop`: must have been duplicate batch,
-      we may be past this L1 block in the safe L2 chain.
+    - `start_epoch_num > epoch.number + 1` -> `future`:
+      i.e. the L1 origin may jump to the next L1 block, but not further.
+    - If `end_epoch_num >= inclusion_block_number` -> `drop`:
+      if the end of the span is past the L1 block it was included in,
+      it cannot possibly reference the chain it was included in, and is thus non-canonical.
+    - If `batch.l1_origin_check` does not match the canonical L1 chain at `end_epoch_num` -> `drop`:
+      verify the batch is intended for this L1 chain.
+    - `start_epoch_num < epoch.number` -> `drop`: must have been duplicate batch,
```

Comment:
Also here.

`start_epoch_num < epoch.number`
becomes
`safe_l2_head.origin.block_number + (origin_changed_bit ? 1 : 0) < safe_l2_head.l1_origin.number`

which can never be satisfied.